File: KmSerializer/Tools/protocol_convertor/template/template_inner.cs
{#- 파일 출력 템플릿 (테스트) -#}
{#- 이하 템플릿 본문 -#}
/**
/// @file {{param['classname'][0] | camel_to_pascal}}.cs
/// @brief 이 파일은 자동생성 되었습니다
*/

using System.Collections;
using System.Collections.Generic;


namespace Yto3.ProtocolService.Param
{

	{%- if param['comment_class'][0] %}
	
	/// <summary>
	/// {{param['comment_class'][0]}}
	/// </summary>
	{%endif-%}
	[System.Serializable]
	public class {{param['classname'][0] | camel_to_pascal}} 
	{
		
		{%- for type in param['type']%}

		{%- if param['comment_member'][loop.index-1] %}
		/// <summary>
		/// {{param['comment_member'][loop.index-1]}}
		/// </summary>
		{%-endif%}
		public {{type}} {{param['member'][loop.index-1]}};
		{%-endfor%}
	}
}

File: KmSerializer/Tools/protocol_convertor/template/template_inner.cpp
{#- 파일 출력 템플릿 -#}

// 자동 생성되는 파일. 직접 수정 금지!
#include "PCH.h"
{%- for i in param['include'] %}
#include "{{i}}"
{%-endfor%}
#include "PktID"
#include "StreamReader.h"
#include "StreamWriter.h"


////////////////////////////////////////////////////////////////////////////////////////////////////
/// @brief	패킷을 직렬화 한다.
////////////////////////////////////////////////////////////////////////////////////////////////////
kmBool Protocol::{{param['classname'][0]}}::Serialize( StreamWriter& writer ) const
{
    {%- for member in param['request_member'] %}
    if( !(writer << {{member}}) ) return false;
    {%- endfor %}

    return true;
}

////////////////////////////////////////////////////////////////////////////////////////////////////
/// @brief	패킷을 역직렬화 한다.
////////////////////////////////////////////////////////////////////////////////////////////////////
kmBool Protocol::{{param['classname'][0]}}::Deserialize( StreamReader& reader )
{
    {%- for member in param['request_member'] %}
    if( !(reader >> {{member}}) ) return false;
    {%- endfor %}

    return true;
}
